### PR TITLE
feat(cli): Madara configuration via environment variables

### PR DESCRIPTION
**Pull request by shamsasari**
They are specified with the prefix `MADARA_` followed by the command-line argument in screaming snake case.

This is part of https://github.com/madara-alliance/madara/issues/285.

### CODE REVIEW

**Comment by Trantorian1** [comment]
Added a quick warning about setting secrets like `MADARA_L1_ENDPOINT` or `MADARA_GATEWAY_KEY` in env. Going forwards (especially considering the support for configuration files in #285) we might want to come up with some better way to have users provide secrets in a safe, isolated way.

**Review by Trantorian1 — APPROVED** [review]
All good :+1:

File: README.md [hunk 1]
```diff
@@ -342,12 +342,18 @@ Toggle details for each namespace to view additional settings:
 
 ### Environment Variables
 
-Set up your node's environment variables using the `STARKNET_` prefix. For example:
+Set up your node's environment variables using the `MADARA_` prefix. For example:
 
-- `STARKNET_BASE_PATH=/path/to/data`
-- `STARKNET_LOG=info`
+- `MADARA_BASE_PATH=/path/to/data`
+- `MADARA_RPC_PORT=1111=1111`
```
Comment (antiyro) [inline README.md]: I think there is a typo here

File: crates/node/src/cli/block_production.rs [hunk 2]
```diff
@@ -3,15 +3,15 @@
 pub struct BlockProductionParams {
     /// Disable the block production service.
     /// The block production service is only enabled with the authority (sequencer) mode.
-    #[arg(long, alias = "no-block-production")]
+    #[arg(env = "MADARA_BLOCK_PRODUCTION_DISABLED", long, alias = "no-block-production")]
     pub block_production_disabled: bool,
 
     /// Launch a devnet with a producton chain id (like SN_MAINNET, SN_SEPOLIA).
```
Comment (antiyro) [inline crates/node/src/cli/block_production.rs]: there is also a typo here on "production"

File: crates/node/src/cli/block_production.rs [hunk 3]
```diff
@@ -3,15 +3,15 @@
...
     /// This in unsafe because your devnet transactiosn can be replayed on the actual network.
```
Comment (antiyro) [inline crates/node/src/cli/block_production.rs]: same here over "transaction"

**Comment by antiyro** [comment]
apart from the typos that are not introduced by you but would be great fixing lgmt!

File: README.md (hunk 1 above)
Comment (shamsasari) [reply README.md]: lol

**Comment by shamsasari** [comment]
Typos fixed.

**Review by antiyro — APPROVED** [review]
lgtm

**Pull request merged**